File: epi_judge_python/is_string_permutable_to_palindrome.py
from test_framework import generic_test
import collections

def can_form_palindrome(s):
    #d = collections.Counter(list(s))
    #oddCtr = 0
    #isPal=True
    #for v in d.values():
    #    if v%2!=0:
    #        oddCtr+=1
    #        if oddCtr>1:
    #            isPal=False
    #            break
    #return isPal

    return sum(v%2 for v in collections.Counter(s).values())<=1
            

    


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main(
            "is_string_permutable_to_palindrome.py",
            'is_string_permutable_to_palindrome.tsv', can_form_palindrome))

File: epi_judge_cpp/is_string_permutable_to_palindrome.cc
#include <algorithm>
#include <unordered_set>
#include <string>
#include "test_framework/generic_test.h"
using std::string;

bool CanFormPalindrome(const string &s)
{

  std::unordered_set<char> set;

  for (char c : s)
  {
    if (set.find(c) == set.end())
    {
      set.insert(c);
    }
    else{
      set.erase(c);
    }
  }
  
  return set.size()<=1;
}

int main(int argc, char *argv[])
{
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"s"};
  return GenericTestMain(args, "is_string_permutable_to_palindrome.cc",
                         "is_string_permutable_to_palindrome.tsv",
                         &CanFormPalindrome, DefaultComparator{}, param_names);
}
